src/kpatch_user.c

```diff
@@ -563,6 +564,87 @@ cmd_update(int argc, char *argv[])
 			  /* send_fd */ -1);
 }
 
+#ifdef STRESS_TEST
+
+struct test_data {
+	int option_period;
+	int stat_cycle_num;
+} test_info = { .option_period = 0, .stat_cycle_num = 0 };
+
+static int
+server_wait(int pid, int period)
+{
+	struct timespec req, rem;
+	int i;
+	req.tv_sec = 0;
+	req.tv_nsec = 1000*1000;
+	for (i=0; i<period; i++) {
+		nanosleep(&req, &rem);
+		if (kill(pid, 0) != 0) {
+			fprintf(stderr, "Process %d terminated.\n", pid);
+			return -1;
+		}
+	}
+	return 0;
+}
+
+static int
+server_stress_test(int argc, char *argv[])
+{
+	int pid;
+	int delay;
+	srand(time(NULL));
+	test_info.stat_cycle_num = 0;
+
+	if (sscanf(argv[1], "%d", &pid) != 1) {
+		kperr("Can't parse pid from %s\n", argv[1]);
+		return -1;
+	}
+
+	while (1) {
+		while (patch_user(storage_dir, pid, 0, -1) < 0)
+			if (server_wait(pid, 1) < 0)
+				return 0;
+		if (test_info.option_period == 0)
+			return 0;
+		delay = rand() % test_info.option_period;
+		if (server_wait(pid, delay) < 0)
+			return 0;
+
+		while (processes_unpatch(pid, 0, 0) < 0)
+			if (server_wait(pid, 1) < 0)
+				return 0;
+		test_info.stat_cycle_num++;
+
+		delay = rand() % test_info.option_period;
+		if (server_wait(pid, delay) < 0)
+			return 0;
+	}
+
+	return 0;
+}
+
+static int cmd_stress_test(int fd, int argc, char *argv[])
+{
+	int child = fork();
+	if (child == 0) {
+		int rv = server_stress_test(argc, argv);
+		exit(rv);
+	}
+
+	int reply=0;
```

Comment:
Why is this required? The child should take care of it, the parent should close the fd.